### PR TITLE
pkg/trace/config: Lower max tracer payload to 25 MB to better align with backend limits

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
Lower the limit for incoming trace payloads to 25MB from 50MB.

### Motivation
There have been a couple customer escalations where they saw errors from the agent indicating a `413` and that the payloads were too large to be accepted by the APM backend. Due to the late nature of these rejections it was difficult to determine the source of these excessively-large traces. By rejecting these payloads earlier it will push the error closer to the true source of the issue (the error will appear at the tracer level and not the agent). It will also improve performance for agents that are receiving these overly-large payloads as the agent will no longer process payloads that would nearly-always be rejected by the backend.

### Possible Drawbacks / Trade-offs
It is unlikely, but possible that _some_ previously accepted and processed payloads will now be rejected. Example: A payload that is 30MB when it enters the agent, but due to a large amount of Tag replacements becomes smaller than the backend limit.
I believe the improved visibility and performance are well worth this risk as trace chunks that are near these limits are difficult to reason about / view in the user-interface. They almost always represent an issue where an integration is connecting work that _should_ be separate traces, or a bug generating a significant number of garbage traces or tags.

### Describe how to test/QA your changes
Send a trace payload that is larger than 25 MB. See that it is rejected by the agent.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/LowerTraceLimit-839236eacc69d87a.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    APM: Lower incoming trace payload limit to 25MB. This more closely aligns with the backend limit. Some users may see traces rejected by the Agent that the agent would have previously accepted, but would have subsequently been rejected by the trace intake.
```
Comment (urseberry) [inline releasenotes/notes/LowerTraceLimit-839236eacc69d87a.yaml:11]:
```suggestion
    APM: Lower incoming trace payload limit to 25MB. This more closely aligns with the backend limit. Some users may see traces rejected by the Agent that the Agent would have previously accepted, but would have subsequently been rejected by the trace intake.
```

File: pkg/trace/config/config.go [hunk 2]
```diff
@@ -464,7 +464,7 @@ func New() *AgentConfig {
 
 		ReceiverHost:           "localhost",
 		ReceiverPort:           8126,
-		MaxRequestBytes:        50 * 1024 * 1024, // 50MB
+		MaxRequestBytes:        25 * 1024 * 1024, // 25MB
```
Comment (ahmed-mez) [inline pkg/trace/config/config.go:467]: Nitpick: We can mention in the comment that `25MB` is the limit enforced by the APM backend.

**Review by ahmed-mez — APPROVED** [review]
LGTM!

One nitpick comment + a couple suggestions about the QA section:

> Send a trace payload that is larger than 25 MB. See that it is rejected by the agent.

Could you mention the expected error log and/or http error code that the agent should return when it receives payloads larger than 25MB?

Also please share the trick to produce these large payloads for testing, it'd be super useful for the person doing QA!

File: pkg/trace/config/config.go (hunk 2 above)
Comment (ajgajg1134) [reply pkg/trace/config/config.go:467]: So the backend limit isn't 25MB (it is intentionally not quite the same and I don't know if we want to make public any exact limit as we could possibly change the limit in the future?)

**Review by urseberry — APPROVED** [review]

**Comment by ajgajg1134** [comment]
> Could you mention the expected error log and/or http error code that the agent should return when it receives payloads larger than 25MB?
> 
> Also please share the trick to produce these large payloads for testing, it'd be super useful for the person doing QA!

Unfortunately I don't have a "trick" I just have a small test program with some code like this that generates BIG traces: 
```
extraSpans := 5_000
for i := 0; i < extraSpans; i++ {
	s, _ := tracer.StartSpanFromContext(sctx, "subSpan")
	bs := make([]byte, 5_000)
	_, e := rand.Read(bs)
	if e != nil {
		fmt.Printf("FAILED TO RANDOM %v\n", e)
	}
	s.SetTag(fmt.Sprintf("SomeKeyWithNumber%d", i), base64.StdEncoding.EncodeToString(bs))
	s.Finish()
}
```

The agent will log something like `Cannot decode v0.4 traces payload: read limit reached` and the tracer sending the request will get a `413` response back. The go tracer logs an error `ERROR: lost 1 traces: payload-too-large`

File: pkg/trace/config/config.go (hunk 2 above)
Comment (ahmed-mez) [reply pkg/trace/config/config.go:467]: Fair point! The comment is for the DD devs who'll read the code in the future, to help them understand the magic number `25MB`, not for users.

**Comment by ahmed-mez** [comment]
> > Could you mention the expected error log and/or http error code that the agent should return when it receives payloads larger than 25MB?
> > Also please share the trick to produce these large payloads for testing, it'd be super useful for the person doing QA!
> 
> Unfortunately I don't have a "trick" I just have a small test program with some code like this that generates BIG traces:
> 
> ```
> extraSpans := 5_000
> for i := 0; i < extraSpans; i++ {
> 	s, _ := tracer.StartSpanFromContext(sctx, "subSpan")
> 	bs := make([]byte, 5_000)
> 	_, e := rand.Read(bs)
> 	if e != nil {
> 		fmt.Printf("FAILED TO RANDOM %v\n", e)
> 	}
> 	s.SetTag(fmt.Sprintf("SomeKeyWithNumber%d", i), base64.StdEncoding.EncodeToString(bs))
> 	s.Finish()
> }
> ```
> 
> The agent will log something like `Cannot decode v0.4 traces payload: read limit reached` and the tracer sending the request will get a `413` response back. The go tracer logs an error `ERROR: lost 1 traces: payload-too-large`

That's super useful, thank you @ajgajg1134 :shipit:

**Pull request merged**